### PR TITLE
chore: extend deploy and upgrade of proxy with deployment type and salt

### PR DESCRIPTION
**Pull request by kiriyaga-txfusion**
# What :computer: 
* Extend deploy and upgrade of proxy with deployment type and salt.

# Why :hand:
* To support different deployment types and salts.

### CODE REVIEW

**Comment by dutterbutter** [comment]
@kiriyaga-txfusion please address the spelling issues here:

```
ome/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:215:1 - Unknown word (Additionaly)
/home/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:215:37 - Unknown word (optionaly)
/home/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:235:28 - Unknown word (ommited)
/home/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:438:1 - Unknown word (Additionaly)
/home/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:438:37 - Unknown word (optionaly)
/home/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:456:28 - Unknown word (ommited)
/home/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:552:1 - Unknown word (Optionaly)
/home/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:706:1 - Unknown word (Optionaly)
/home/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:915:28 - Unknown word (ommited)
```

**Comment by kiriyaga-txfusion** [comment]
> @kiriyaga-txfusion please address the spelling issues here:
> 
> ```
> ome/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:215:1 - Unknown word (Additionaly)
> /home/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:215:37 - Unknown word (optionaly)
> /home/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:235:28 - Unknown word (ommited)
> /home/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:438:1 - Unknown word (Additionaly)
> /home/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:438:37 - Unknown word (optionaly)
> /home/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:456:28 - Unknown word (ommited)
> /home/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:552:1 - Unknown word (Optionaly)
> /home/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:706:1 - Unknown word (Optionaly)
> /home/runner/work/zksync-web-era-docs/zksync-web-era-docs/docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:915:28 - Unknown word (ommited)
> ```

Resolved

**Review by dutterbutter — DISMISSED** [review]

File: docs/build/tooling/hardhat/hardhat-zksync-upgradable.md [hunk 1]
```diff
@@ -212,6 +212,29 @@ npx hardhat run SCRIPT_FILE
...
+```
+
+:::note
+Permissible values for the deployment type include `create`, `create2`, `createAccount`, and `create2Account`. If this parameter is omitted, the default value will be `create`.
+If the salt parameters are omitted, the default value will be `0x0000000000000000000000000000000000000000000000000000000000000000`.
+:::
+
 ### Openzeppelin Version
 
 The plugin does not work with the latest versions due to a blocker on the `@matterlab/zksync-contracts` package.The solution is to change the development dependencies to the previous version in your `package.json`.
```
Comment (uF4No) [inline docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:240]: Can we remove this section and merge it with the warning in the "Installation" section at the beginning of the page?

File: docs/build/tooling/hardhat/hardhat-zksync-upgradable.md [hunk 2]
```diff
@@ -412,6 +435,27 @@ npx hardhat run SCRIPT_FILE
...
+  deploymentType: 'create2',
+  salt: '0x6273795673417857416686492163276941983664248508133571812215241323'
+});
+await box.waitForDeployment();
+```
+
+:::note
+Permissible values for the deployment type include `create`, `create2`, `createAccount`, and `create2Account`. If this parameter is omitted, the default value will be `create`.
+If the salt parameters are omitted, the default value will be `0x0000000000000000000000000000000000000000000000000000000000000000`.
+:::
```
Comment (uF4No) [inline docs/build/tooling/hardhat/hardhat-zksync-upgradable.md:457]:
```suggestion
:::info
Allowed values for the deployment type include `create`, `create2`, `createAccount`, and `create2Account`. If this parameter is omitted, the default value will be `create`.
If the salt parameters are omitted, the default value will be `0x0000000000000000000000000000000000000000000000000000000000000000`.
:::
```

**Review by uF4No — APPROVED** [review]

**Review by uF4No — APPROVED** [review]

**Pull request merged**